packages/core/src/attestation/Attestation.ts

```diff
@@ -32,245 +24,167 @@ import { DelegationNode } from '../delegation/DelegationNode.js'
-   */
-  public async getRemoveTx(maxDepth: number): Promise<SubmittableExtrinsic> {
-    return getRemoveTx(this.claimHash, maxDepth)
+  if (typeof input === 'string') {
+    delegationId = input
+  } else {
+    delegationId = input.delegationId
   }
 
-  /**
-   * [STATIC] [ASYNC] Reclaims the deposit of an attestation and removes the attestation. Also available as an instance method.
-   *
-   * This call can only be successfully executed if the submitter of the transaction is the original payer of the attestation deposit.
-   *
-   * @returns A promise containing the unsigned SubmittableExtrinsic (submittable transaction).
-   */
-  public async getReclaimDepositTx(): Promise<SubmittableExtrinsic> {
-    return getReclaimDepositTx(this.claimHash)
+  if (!delegationId) {
+    return null
   }
 
-  /**
-   * [STATIC] [ASYNC] Queries an attestation from the chain and checks its validity.
-   *
-   * @param attestation - The Attestation to verify.
-   * @param claimHash - The hash of the claim that corresponds to the attestation to check. Defaults to the claimHash for the attestation onto which "verify" is called.
-   * @returns A promise containing whether the attestation is valid.
-   */
-  public static async checkValidity(
-    attestation: IAttestation,
-    claimHash: IAttestation['claimHash'] = attestation.claimHash
-  ): Promise<boolean> {
-    // Query attestation by claimHash. null if no attestation is found on-chain for this hash
-    const chainAttestation: Attestation | null = await Attestation.query(
-      claimHash
-    )
-    return !!(
-      chainAttestation !== null &&
-      chainAttestation.owner === attestation.owner &&
-      !chainAttestation.revoked
-    )
+  const delegationNode = await DelegationNode.query(delegationId)
+  if (!delegationNode) {
+    return null
   }
+  return delegationNode.getHierarchyDetails()
+}
 
-  public async checkValidity(): Promise<boolean> {
-    return Attestation.checkValidity(this)
+/**
+ * Custom Type Guard to determine input being of type IAttestation.
+ *
+ * @param input The potentially only partial IAttestation.
+ * @returns Boolean whether input is of type IAttestation.
+ */
+export function isIAttestation(input: unknown): input is IAttestation {
+  try {
+    verifyDataStructure(input as IAttestation)
+  } catch (error) {
+    return false
   }
+  return true
+}
 
-  /**
-   * Compresses an [[Attestation]] object.
-   *
-   * @returns An array that contains the same properties of an [[Attestation]].
-   */
-  public compress(): CompressedAttestation {
-    return AttestationUtils.compress(this)
-  }
+/**
+ * Queries an attestation from the chain and checks if it is existing, if the owner of the attestation matches and if it was not revoked.
+ *
+ * @param attestation - The Attestation to verify.
+ * @param claimHash - The hash of the claim that corresponds to the attestation to check. Defaults to the claimHash for the attestation onto which "verify" is called.
```

Comment:
I guess there is some reason to provide it separately, but this is so weird!

Reply:
very good question in which scenario that would be beneficial